lang/LangPrimSource/PyrStringPrim.cpp

```diff
@@ -622,10 +622,13 @@ int prString_Getenv(struct VMGlobals* g, int /* numArgsPushed */) {
         return this_err;
 
 #ifdef _WIN32
-    const auto count = GetEnvironmentVariable(this_str.c_str(), nullptr, 0);
-    std::string buf(count, 0);
+    const auto this_str_w = SC_Codecvt::utf8_cstr_to_utf16_wstring(this_str.c_str());
+    const auto count = GetEnvironmentVariableW(this_str_w.c_str(), nullptr, 0);
+    std::string buf;
     if (count != 0) {
-        GetEnvironmentVariable(this_str.c_str(), buf.data(), buf.size());
+        std::wstring wbuf(count, 0);
+        GetEnvironmentVariableW(this_str_w.c_str(), &wbuf[0], count);
```

Comment:
`&wbuf[0]` could be `wbuf.data()`.

Reply:
I was a bit confused whether string.data() should be allowed to be manipulated/modified or not... I'm happy to change, just trying to understand the nuances.

Reply:
The non-const overload of `data` was added in c++17 (after strings became contiguous in 11).

Cpp ref mentions that it is undefined behaviour to change the char at `data() + size()` to anything other than the null terminator. 

I'd say, the presence of a non-const overload (that isn't there due to legacy reasons) and cpp ref's warning means mutations are fine.

Reply:
Thanks for the explanation!